### ISSUE TITLE
ERROR: for bind9-mailcow  Container "5dfe741dcd7c" is unhealthy.

### ISSUE THREAD

**Issue by wpbeaches** [mailcow/mailcow-dockerized#252]
Spinning up a new Vultr Ubuntu instance - on install I receive this error

```
root@wp-beaches-noserverpilot-la:~/mailcow-dockerized# docker-compose up -d
mailcowdockerized_mysql-mailcow_1 is up-to-date
mailcowdockerized_clamd-mailcow_1 is up-to-date

ERROR: for bind9-mailcow  Container "5dfe741dcd7c" is unhealthy.
ERROR: Encountered errors while bringing up the project.
```

**Comment by broedli** [comment]
Please make sure you have:
- The latest docker images for mailcow: `docker-compose pull`
- The latest version of mailcow: `git pull`
- A recent version of docker `docker -v` (17.03) and docker-compose `docker-compose -v` (1.12)
- **Provided logs to your issue for your failing container and it's dependencies**: `docker-compose logs --tail=100 bind9-mailcow`**

The status *unhealthy* comes from a failing [healthcheck](https://github.com/mailcow/mailcow-dockerized/blob/069c93cb64308f0259658ff7eb0d867085f425eb/docker-compose.yml#L21-L25) of the containers [dependencies](https://github.com/mailcow/mailcow-dockerized/blob/069c93cb64308f0259658ff7eb0d867085f425eb/docker-compose.yml#L7-L9). In bind9's case that means: mysql-mailcow. Can you start this container by itself? `docker-compose start mysql-mailcow` Can all containers contact each other over the network 172.22.1.1/24?

Lastly make sure you search for your problem first, before you create an issue:

Please have a look at #91, #114 and #156

**Comment by buletta** [comment]
I had the same error running docker as root on photon OS. Try changing `~/mailcow-dockerized/data/conf/mysql` to 755. Check your umask setting. On photon it was set 027. Try setting umask to 022 before you git pull.

**Comment by jpbaril** [comment]
I tried to try mailcow for the first time. I also have the exact same error and indeed the unhealthy container id is of the mariadb container. I deleted the MySQL/MariaDB container and image then tried buletta suggestion and finally pulled image again and compose up. That did not work. Maybe I did do it correctly.

I have the docker and composer version broedli mentions.

I tried "docker-compose start mysql-mailcow" and it said "Starting mysql-mailcow ... done"

Here are logs of mysql-mailcow
`mysql-mailcow_1      | 2017-06-08  9:05:15 140134046193600 [Note] InnoDB: The InnoDB memory heap is disabled
mysql-mailcow_1      | 2017-06-08  9:05:15 140134046193600 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mysql-mailcow_1      | 2017-06-08  9:05:15 140134046193600 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
mysql-mailcow_1      | 2017-06-08  9:05:15 140134046193600 [Note] InnoDB: Compressed tables use zlib 1.2.8
mysql-mailcow_1      | 2017-06-08  9:05:15 140134046193600 [Note] InnoDB: Using Linux native AIO
mysql-mailcow_1      | 2017-06-08  9:05:15 140134046193600 [Note] InnoDB: Using SSE crc32 instructions
mysql-mailcow_1      | 2017-06-08  9:05:15 140134046193600 [Note] InnoDB: Initializing buffer pool, size = 256.0M
mysql-mailcow_1      | InnoDB: mmap(281149440 bytes) failed; errno 12
mysql-mailcow_1      | 2017-06-08  9:05:15 140134046193600 [ERROR] InnoDB: Cannot allocate memory for the buffer pool
mysql-mailcow_1      | 2017-06-08  9:05:15 140134046193600 [ERROR] Plugin 'InnoDB' init function returned error.
mysql-mailcow_1      | 2017-06-08  9:05:15 140134046193600 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
mysql-mailcow_1      | 2017-06-08  9:05:15 140134046193600 [Note] Plugin 'FEEDBACK' is disabled.
mysql-mailcow_1      | 2017-06-08  9:05:15 140134046193600 [ERROR] Unknown/unsupported storage engine: InnoDB
mysql-mailcow_1      | 2017-06-08  9:05:15 140134046193600 [ERROR] Aborting
`

And I don't know how to verify that all containers can contact each other over the network 172.22.1.1/24 as I'm totally new to docker and not that familiar with networks and firewalls on Linux, especially through command line.

Thanks.

**Comment by chriscroome** [comment]
> `2017-06-08 9:05:15 140134046193600 [Note] InnoDB: Initializing buffer pool, size = 256.0M mysql-mailcow_1 | InnoDB: mmap(281149440 bytes) failed; errno 12 mysql-mailcow_1`
> `2017-06-08 9:05:15 140134046193600 [ERROR] InnoDB: Cannot allocate memory for the buffer pool mysql-mailcow_1`

Do you have enough RAM available?

**Comment by stevesbrain** [comment]
^ Was going to ask the same question. If unsure, maybe watch it with ```htop``` as you try to start the container - see whether it uses it all up

**Comment by jpbaril** [comment]
Thanks for your answers. Well it 'solved' it. I had allocated 1 GB of memory which I thought would be enough. I allocated 1.5 Gb and now it _ups_ (?). But now I have this error message instead of webui in browser: "Host '172.22.1.9' is not allowed to connect to this MariaDB server". I'm on RancherOS. I also tried with fedora-atomic but got "mysqld: Can't read dir of '/etc/mysql/conf.d/'" in logs when trying to 'docker-compose up'. I'll continue these issue discussions in the forum instead:

- https://forum.mailcow.email/t/host-172-22-1-9-is-not-allowed-to-connect-to-this-mariadb-server/587
- https://forum.mailcow.email/t/mysqld-cant-read-dir-of-etc-mysql-conf-d/586
